### ISSUE TITLE
Running py.test throws module errors in fresh clone of the project

### ISSUE THREAD

**Issue by jashgala** [dbcli/athenacli#22]
As per the documentation, after pulling the requirements-dev.txt using pip, running `py.test` should run the tests.

However, running `py.test` throws Module errors in the terminal. See the ascii cast below:

[![asciicast](https://asciinema.org/a/evRJYgSASqPe8lLkBmWyOi4tY.png)](https://asciinema.org/a/evRJYgSASqPe8lLkBmWyOi4tY)

[Download log.txt](https://github.com/dbcli/athenacli/files/2467430/log.txt)
<details>
<summary>Click to see log here!</summary>
<p>

```
============================= test session starts ==============================
platform darwin -- Python 3.6.5, pytest-3.8.2, py-1.6.0, pluggy-0.7.1
rootdir: /Users/jashgala/personal/Github/athenacli, inifile:
collected 478 items / 11 errors

==================================== ERRORS ====================================
___ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_commands.py ____
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_commands.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
_____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_core.py ______
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_core.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
_____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_files.py _____
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_files.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
_____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_hooks.py _____
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_hooks.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
__ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_integration.py __
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_integration.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
___ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_packaging.py ___
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_packaging.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
___ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_pbr_json.py ____
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_pbr_json.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
_____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_setup.py _____
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_setup.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
_____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_util.py ______
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_util.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_version.py ____
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_version.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
_____ ERROR collecting lib/python3.6/site-packages/pbr/tests/test_wsgi.py ______
ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_wsgi.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/pbr/tests/__init__.py:16: in <module>
    import testscenarios
E   ModuleNotFoundError: No module named 'testscenarios'
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
=========================== 11 error in 8.32 seconds ===========================
```
</p>
</details>

<br />

**Workaround:** Running `pytest test` seems to be working for the time being

**Comment by zzl0** [comment]
`ImportError while importing test module '/Users/jashgala/personal/Github/athenacli/lib/python3.6/site-packages/pbr/tests/test_commands.py'.`

it seems you have a `lib` directory under your `/Users/jashgala/personal/Github/athenacli/` directory, so py.test tried to run the tests of `lib`, which are not `athenacli`'s tests.

It's because you created venv in a "wrong" way. Please try below commands:

``` bash
$ git clone https://github.com/dbcli/athenacli.git
$ cd athenacli
$ virtualenv -p python3 venv
$ source venv/bin/activate
$ pip install -e .  # install athenacli
$ pip install -r requirements-dev.txt # install test requirments
```

I usually use `pipenv` to create a virtualenv now (`pipenv shell`), you can give a try.

**Comment by jashgala** [comment]
I believe the reason there's a lib directory is that instead of creating a virtual environment inside the root of my folder, I am creating one in the parent directory with the same name as the root folder. Still need to get my head around the best practice for creating virtual envs.

BTW, the command I used `python -m venv` is the new official python way of virtual env creation (since they integrated it as a language feature in 3.x).

The issue disappears after I created a venv in a different directory. Closing this!
